src/main/java/com/salesforce/storm/spout/dynamic/coordinator/SpoutMonitor.java

```diff
@@ -215,15 +220,22 @@ public void run() {
                 // Pause for a period before checking for more spouts
                 try {
                     Thread.sleep(getMonitorThreadIntervalMs());
-                } catch (InterruptedException ex) {
+                } catch (final InterruptedException ex) {
                     logger.warn("Thread interrupted, shutting down...");
                     return;
                 }
             }
             logger.warn("Spout coordinator is ceasing to run due to shutdown request...");
-        } catch (Exception ex) {
-            // TODO: Should we restart the monitor?
-            logger.error("!!!!!! SpoutMonitor threw an exception {}", ex);
+        } catch (final Exception ex) {
+            // We handle restarting spout monitor in the coordinator, which is who monitors this thread.
+            // Lets report the error
+            reportError(ex);
+
+            // Log it.
+            logger.error("!!!!!! SpoutMonitor threw an exception {}", ex.getMessage(), ex);
```

Comment:
I've never been a big fan of this log line. Probably just drop the exclamation points IMO.